### PR TITLE
Feat windows scripts

### PR DESCRIPTION
**Pull request by george-haddad**
This pull request addresses support for Windows and Ubuntu Linux

- Add support for Ubuntu Linux
- Add support for Windows, which is using the Ubuntu bash under Linux Subsystem
- Unified the mysql server installation to use `root` at default password
  - On Linux mysql client will not allow you to connect directly without specifying a password
- Add yarn install shortcuts for mac, windows and ubuntu
- Update the Readme installation script

This was tested on Windows 10 Pro 64bit Anniversary edition with Linux Subsystem enabled with Ubuntu bash (Ubuntu Xenial). Was also tested on a Mac Book Pro with OS High Sierra v10.13.

This PR is related to issue #12

### CODE REVIEW

File: RoboFile.php [hunk 1]
```diff
@@ -26,14 +26,14 @@ public function wordpressSetup( $opts = [
...
+		$this->_exec( "mysql -uroot -proot -h 0.0.0.0 -e 'create user if not exists " . $opts['wp-db-name'] . "'" );
+		$this->_exec( "mysql -uroot -proot -h 0.0.0.0 -e 'create database if not exists " . $opts['wp-db-name'] . "'" );
+		$this->_exec( 'mysql -uroot -proot -h 0.0.0.0 -e "grant all privileges on ' . $opts['wp-db-name']
 		. ' . * to ' . $opts['wp-db-name'] . "@localhost identified by '" . $opts['wp-db-name'] . "'\"" );
-		$this->_exec( "mysql -uroot -e 'flush privileges'" );
+		$this->_exec( "mysql -uroot -proot -h 0.0.0.0 -e 'flush privileges'" );
 
-		$this->wp( 'core download --version=4.8.2 --force' );
-		$this->wp( 'core config --dbname=' . $opts['wp-db-name'] . ' --dbuser=' . $opts['wp-db-name'] . ' --dbpass=' . $opts['wp-db-name'] . ' --dbhost=127.0.0.1' );
+		$this->wp( 'core download --version=4.8.2 --locale=en_GB --force' );
```
Comment (ginatrapani) [inline RoboFile.php]: Could we make this locale=en_US? And then .gitignore `wordpress/wp-content/languages/`?

Comment (george-haddad) [reply RoboFile.php]: There is a documented error with the en_US locale here  https://github.com/wp-cli/wp-cli/issues/4370 which is why I used the en_GB. I couldn't find a way to resolve the issue without choosing another locale.

Comment (ginatrapani) [reply RoboFile.php]: Ah, I see! okay then, `en_GB` it is.

**Comment by george-haddad** [comment]
Updated the branch to include the following:

- OS auto-detection and run appropriate script
- Add language dir to .gitingore
- Using default en_US locale
- Test on Ubuntu shell for windows and Mac OS X

There still exists the warnings for the locale but it seems to work with it. You might want to clear your `.wp-clie/cache` folder before running the test again.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        chmod +x ./scripts/ubuntu.sh
+        ./scripts/ubuntu.sh
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+        chmod +x ./scripts/macosx.sh
+        ./scripts/macosx.sh
+else
+        echo "Unhandled OS type"
```
Comment (ginatrapani) [inline install.sh]: This looks great! Let's update this message to something more conversational, like "Sorry, this installation script only works on Mac OS X and Ubuntu Linux. Looks like your operating system is $OSTYPE."

File: package.json [hunk 3]
```diff
@@ -6,9 +6,7 @@
   "license": "UNLICENSED",
   "private": true,
   "scripts": {
-    "ubuntu": "chmod +x ./scripts/ubuntu.sh && ./scripts/ubuntu.sh",
-    "windows": "chmod +x ./scripts/ubuntu.sh && ./scripts/ubuntu.sh",
-    "macosx": "chmod +x ./scripts/macosx.sh && ./scripts/macosx.sh",
+    "preinstall": "chmod +x install.sh && ./install.sh",
```
Comment (ginatrapani) [inline package.json]: Excellent! Let's update the README to reflect that the command is just `yarn install && yarn start` now.

File: install.sh (hunk 2 above)
Comment (george-haddad) [reply install.sh]: That is friendlier :)

File: package.json (hunk 3 above)
Comment (george-haddad) [reply package.json]: Changes done.

**Pull request merged**